packages/no-trapping-shim/src/no-trapping-pony.js

```diff
@@ -0,0 +1,278 @@
+  } else {
+    return apply(OriginalObject, this, args);
+  }
+};
+// @ts-expect-error We actually can assign to its `.prototype`.
+ObjectPlus.prototype = OriginalObject.prototype;
+addExtras(ObjectPlus, OriginalObject, extraObjectMethods);
+export { ObjectPlus };
+
+const metaHandler = freeze({
+  get(_, trapName, handlerPlus) {
+    /**
+     * The `trapPlus` method is an enhanced version of
+     * `originalHandler[trapName]`. If the handlerPlus has no own `trapName`
+     * property, then the `get` of the metaHandler is called, which returns
+     * the `trapPlus`, which is then called as the trap of the returned
+     * proxyPlus. When so called, it installs an own `handlerPlus[trapName]`
+     * which is either `undefined` or this same `trapPlus`, to avoid further
+     * need to meta-handle that `handlerPlus[trapName]`.
+     *
+     * @param {any} target
+     * @param {any[]} rest
+     */
+    const trapPlus = freeze((target, ...rest) => {
+      if (isNoTrappingInternal(target, true)) {
+        defineProperty(handlerPlus, trapName, {
+          value: undefined,
+          writable: false,
+          enumerable: true,
+          configurable: false,
+        });
+      } else {
+        if (!hasOwn(handlerPlus, trapName)) {
+          defineProperty(handlerPlus, trapName, {
+            value: trapPlus,
+            writable: false,
+            enumerable: true,
+            configurable: true,
+          });
+        }
+        const { originalHandler } = handlerPlus;
+        const trap = originalHandler[trapName];
+        if (trap !== undefined) {
+          // Note that whether `trap === undefined` can change dynamically,
+          // so we do not install an own `handlerPlus[trapName] === undefined`
+          // for that case. We still install or preserve an own
+          // `handlerPlus[trapName] === trapPlus` until the target is
+          // seen to be non-trapping.
+          return apply(trap, originalHandler, [target, ...rest]);
+        }
+      }
+      return ReflectPlus[trapName](target, ...rest);
+    });
+    return trapPlus;
+  },
+});
+
+/**
+ * A handlerPlus starts as a fresh empty object that inherits from a proxy
+ * whose handler is the shared generic metaHandler.
+ * Thus, the metaHandler's `get` method is called only when the
+ * `handlerPlus` does not have a property overriding that `trapName`.
+ * In that case, the metaHandler's `get` is called with its `receiver`
+ * being the `handlerPlus`.
+ *
+ * @param {ProxyHandler<any>} originalHandler
+ * @returns {ProxyHandler<any> & {
+ *   isNoTrapping: (target: any) => boolean,
+ *   suppressTrapping: (target: any) => boolean,
+ *   originalHandler: ProxyHandler<any>
+ * }}
+ */
+const makeHandlerPlus = originalHandler => ({
+  // @ts-expect-error TS does not know what this __proto__ is doing
+  __proto__: new OriginalProxy({}, metaHandler),
+  // relies on there never being a trap named `originalHandler`.
+  originalHandler,
```

Comment:
To avoid any potential collisions with trap methods, let's store the original handler as a unique symbol property on `handlerPlus`. After all that's what symbols were meant for :)

```js
const ORIGINAL_HANDLER = Symbol('OriginalHandler');

const makeHandlerPlus = originalHandler => ({
  __proto__: new OriginalProxy({}, metaHandler),
  [ORIGINAL_HANDLER]: originalHandler,
});
```

Reply:
Good idea. Done